assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -24,12 +24,24 @@ import { useEffect, useCallback } from 'react';
  * Internal dependencies
  */
 import { useAPI, useConfig } from '../';
+import { useUploader } from '../uploader';
+import { useStory } from '../story';
+import {
+  getResourceFromLocalFile,
+  getResourceFromUploadAPI,
+} from '../../app/media/utils';
+import useInsertElement from '../../components/canvas/useInsertElement';
```

Comment:
This is not available on this level, is it? IMHO, it'd be easier to just pass `insertElement` function as a callback to the `uploadMediaFromWorkspace`.